allensdk/brain_observatory/behavior/data_objects/timestamps/stimulus_timestamps/timestamps_processing.py

```diff
@@ -47,3 +47,59 @@ def get_ophys_stimulus_timestamps(sync_path: Union[str, Path]) -> np.ndarray:
     aligner = OphysTimeAligner(sync_file=sync_path)
     stimulus_timestamps, _ = aligner.clipped_stim_timestamps
     return stimulus_timestamps
+
+
+def get_frame_indices(
+        frame_timestamps: np.ndarray,
+        event_timestamps: np.ndarray) -> np.ndarray:
+    """
+    Given an array of timestamps corresponding to stimulus frames
+    and an array of timestamps corresponding to some event (i.e.
+    licks), return an array of indexes indicating which frame
+    each event occured on. Indexes will be chosen to be the
+    smallest index satisfying
+
+    frame_timestamps[event_indices] >= event_timestamps
+
+    Parameters
+    ----------
+    frame_timestamps: np.ndarray
+        must be in ascending order
+
+    event_timetamps: np.ndarray
+
+    Returns
+    -------
+    event_indices: np.ndarray
+        integers that are all >= 0 and < len(frame_timestamps)
+
+    Note
+    ----
+    If a value is repeated in frame_timestamps, the first suitable
+    frame index is returned for a given event_timestamp
+    """
+
+    delta_t = np.diff(frame_timestamps)
+    if delta_t.min() < -1.0e-10:
+        raise ValueError("frame_timestamps are not in ascending order")
+
+    n_frames = len(frame_timestamps)
+
+    event_indices = np.searchsorted(
+                        frame_timestamps,
+                        event_timestamps,
+                        side='left')
+
+    event_indices = np.clip(event_indices, None, n_frames-1)
+
+    # correct for fact that searchsorted will select as
+    # frame index the first frame time that is larger
+    # than lick_times; we want the lick_times associated
+    # with the last frame that is smaller than the lick_time
```

Comment:
Can't this be done by setting `side='right'` instead? The numpy reference claims that 'right' returns the "return the last such index."

Reply:
It can't. The docs say
```
right  ``a[i-1] <= v < a[i]``
```
and I want
```
a[i] <= v < a[i+1]
```
i.e. I want the lick_time to be after the start of the assigned frame and before the start of the next frame.

I think I wrote my docstring wrong, though. Will fix.